### PR TITLE
Add the .gitignore file

### PR DESCRIPTION
**Pull request by shaktipn**
Removed the already committed files which were supposed to be ignored.
**Then added the .gitignore file at the root** and verified that it was working by creating a file that will match the ignore pattern which wasn't being tracked by git as expected.

### CODE REVIEW

File: .idea/.gitignore
Comment (sudarshan-dodiya) [inline .idea/.gitignore:1]: Please make sure .idea folder is gitignored as well. This is a folder IntelliJ automatically creates when a project in opened, and exists mostly for personal IDE configuration. This should not be committed, and if it is committed, should be done with proper thought behind it.

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+*folder[123]
+*file?.txt
+
```
Comment (sudarshan-dodiya) [inline .gitignore:3]: Why this extra line?

File: exercises/creating-a-pull-request/ABOUT_ME.md
Comment (sudarshan-dodiya) [inline exercises/creating-a-pull-request/ABOUT_ME.md:1]: Why does this file exist in this PR? Please make sure that your PR only contains changes pertaining to that PR, and not from other PRs. Otherwise reviewing stuff will become a nightmare.

File: exercises/git-ignore/files/file1.txt [hunk 2]
```diff
@@ -1 +0,0 @@
-Do not gitginore this file.
```
Comment (sudarshan-dodiya) [inline exercises/git-ignore/files/file1.txt:1]: This says that this file should not be gitignored, why is this ignored then?

File: exercises/git-ignore/files/file2.txt [hunk 3]
```diff
@@ -1 +0,0 @@
-Ignore this file, but the rule should not show up in project gitignore
```
Comment (sudarshan-dodiya) [inline exercises/git-ignore/files/file2.txt:1]: Do you know how to do this?

**Review by sudarshan-dodiya — CHANGES_REQUESTED** [review]
Again, the PR title and commit history does not conform to a good format, please make sure that you follow proper conventions as described in the docs.

File: .idea/.gitignore
Comment (shaktipn) [reply .idea/.gitignore:1]: Got it

File: .gitignore (hunk 1 above)
Comment (shaktipn) [reply .gitignore:3]: I removed the extra line

File: exercises/creating-a-pull-request/ABOUT_ME.md
Comment (shaktipn) [reply exercises/creating-a-pull-request/ABOUT_ME.md:1]: I have reviewed the PR and removed the ABOUT_ME.md from this PR.

File: exercises/git-ignore/files/file1.txt (hunk 2 above)
Comment (shaktipn) [reply exercises/git-ignore/files/file1.txt]: I have updated the pattern in .gitignore . Now its not ignored.

File: exercises/git-ignore/files/file2.txt (hunk 3 above)
Comment (shaktipn) [reply exercises/git-ignore/files/file2.txt:1]: Yeah I added it to the exclude file in .git

File: exercises/git-ignore/files/folder3/file35.txt [hunk 4]
```diff
@@ -1 +0,0 @@
-Do not ignore this file
```
Comment (sudarshan-dodiya) [inline exercises/git-ignore/files/folder3/file35.txt:1]: Why is this file still deleted? Please make sure to review your own PRs before asking for a review.

**Review by sudarshan-dodiya — CHANGES_REQUESTED** [review]

File: exercises/git-ignore/files/folder3/file35.txt (hunk 4 above)
Comment (shaktipn) [reply exercises/git-ignore/files/folder3/file35.txt]: I modified the .gitignore pattern to not ignore it anymore.

**Review by sudarshan-dodiya — APPROVED** [review]
